File: backend/dogadoption/src/main/java/com/example/dogadoption/services/AdoptionInterestService.java
package com.example.dogadoption.services;

import com.example.dogadoption.models.AdoptionInterest;

import java.util.List;
import java.util.Optional;

public interface AdoptionInterestService {
    AdoptionInterest createAdoption(AdoptionInterest adoptionInterest);

    List<AdoptionInterest> findUserAdoption(List<String> id);

    List<AdoptionInterest> adoptionInterestList();

    Optional<AdoptionInterest> findAdoptionInterest(String id);

    AdoptionInterest updateAdoptionInterestStatus(String id, AdoptionInterest adoptionInterest);

    void deleteAdoptionInterest(String id);
}

File: frontend/app/src/app/services/adoption-interest.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable, catchError } from 'rxjs';
import { AdoptionInterest } from '../models/adoption-interest';

@Injectable({
  providedIn: 'root'
})
export class AdoptionInterestService {
  appUrl: string;

  constructor(private http: HttpClient) {
    this.appUrl = "http://localhost:18080"
  }
  
  public getAdoptionList(): Observable<AdoptionInterest[]> {
    return this.http.get<AdoptionInterest[]>(this.appUrl + "/adoption/list");
  }

  public createAdoption(adoptionForm: AdoptionInterest){
    return this.http.post(this.appUrl + "/adoption/create", adoptionForm).subscribe(
      (response) =>{
        console.log("Dog created: ", response);
      },
      (error) => {
        console.error("Error Dog creation: ", error)
      }
    )
  }

  public findAdoptionInterest(id: Number){
    return this.http.get<AdoptionInterest>(this.appUrl + "/adoption/" + id.toString());
  }

  public deleteAdoptionInterest(id: Number){
    return this.http.get<AdoptionInterest>(this.appUrl + "/adoption/delete/" + id.toString())
  }
}
